experimental/packages/opentelemetry-sdk-metrics-base/src/state/TemporalMetricProcessor.ts

```diff
@@ -152,19 +145,45 @@ export class TemporalMetricProcessor<T> {
     return result;
   }
 
-  static merge<T>(last: AttributeHashMap<T>, current: AttributeHashMap<T>, aggregator: Aggregator<T>) {
+  static merge<T extends Maybe<Accumulation>>(last: AttributeHashMap<T>, current: AttributeHashMap<T>, aggregator: Aggregator<T>) {
     const result = last;
     const iterator = current.entries();
     let next = iterator.next();
     while (next.done !== true) {
       const [key, record, hash] = next.value;
-      const lastAccumulation = last.get(key, hash) ?? aggregator.createAccumulation();
-      result.set(key, aggregator.merge(lastAccumulation, record), hash);
+      if (last.has(key, hash)) {
+        const lastAccumulation = last.get(key, hash);
+        // lastAccumulation must present.
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const accumulation = aggregator.merge(lastAccumulation!, record);
+        result.set(key, accumulation, hash);
+      } else {
+        result.set(key, record, hash);
+      }
 
       next = iterator.next();
     }
     return result;
   }
+
+  /**
+   * Calibrate the reported metric streams's startTime to lastCollectionTime. Leaves
+   * the new stream to be the initial observation time unchanged.
+   */
+  static calibrateStartTime<T extends Maybe<Accumulation>>(last: AttributeHashMap<T>, current: AttributeHashMap<T>, lastCollectionTime: HrTime) {
+    const iterator = last.keys();
+    let next = iterator.next();
+    while (next.done !== true) {
+      const [key, hash] = next.value;
+      if (current.has(key, hash)) {
+        const currentAccumulation = current.get(key, hash);
+        currentAccumulation?.setStartTime(lastCollectionTime);
+      }
+
+      next = iterator.next();
+    }
```

Comment:
why manually iterating instead of using a for..of loop? I think you should be able to do:

```suggestion
    for (const [key, hash] of last.keys()) {
    	current.get(key, hash)?.setStartTime(lastCollectionTime);
    }
```